### PR TITLE
Support lang parameter for Koncorde filters

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

Since Kuzzle 2.8.0 supports the "lang" parameter to handle Konkorde filters, we need to add it in SDKs for the API methods who accept it.

https://github.com/kuzzleio/sdk-jvm/issues/40

### Boyscout

Fix doc for mUpdate and updateByQuery

### CODE REVIEW

File: doc/1/controllers/auth/get-current-user/index.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ description: Returns the profile object for the user linked to the `JSON Web Tok
 
 # GetCurrentUserAsync
 
-Returns informations about the user currently loggued with the SDK instance.
+Returns informations about the user currently logged with the SDK instance.
```
Comment (Leodau) [inline doc/1/controllers/auth/get-current-user/index.md:10]: “information” is an uncountable noun. :eyes: 
```suggestion
Returns information about the user currently logged with the SDK instance.
```

File: doc/1/controllers/auth/search-api-keys/index.md [hunk 2]
```diff
@@ -13,6 +13,19 @@ description: Searches API keys for the currently logged user.
...
 
+Kuzzle uses the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
+
+<SinceBadge version="change-me"/>
+
+This method also supports the [Koncorde Filters DSL](/core/2/api/koncorde-filters-syntax) to match documents by passing the `lang` argument with the value `koncorde`.  
+Koncorde filters will be translated into an Elasticsearch query.  
+
+::: warning
+Koncorde `bool` operator and `regexp` clause are not supported for search queries.
```
Comment (Leodau) [inline doc/1/controllers/auth/search-api-keys/index.md:24]:
```suggestion
Koncorde `bool` operator and `regexp` clauses are not supported for search queries.
```

File: doc/1/controllers/document/search/index.md [hunk 3]
```diff
@@ -28,6 +28,15 @@ It can lead to memory leaks if a scroll duration too great is provided, or if to
 You can restrict the scroll session maximum duration under the `services.storage.maxScrollDuration` configuration key.
 :::
 
+<SinceBadge version="change-me"/>
+
+This method also supports the [Koncorde Filters DSL](/core/2/api/koncorde-filters-syntax) to match documents by passing the `lang` argument with the value `koncorde`.  
+Koncorde filters will be translated into an Elasticsearch query.  
+
+::: warning
+Koncorde `bool` operator and `regexp` clause are not supported for search queries.
```
Comment (Leodau) [inline doc/1/controllers/document/search/index.md:37]:
```suggestion
Koncorde `bool` operator and `regexp` clauses are not supported for search queries.
```

File: doc/1/controllers/document/update-by-query/index.md [hunk 4]
```diff
@@ -11,6 +11,15 @@ Updates documents matching the provided search query.
 
 Kuzzle uses the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
 
+<SinceBadge version="change-me"/>
+
+This method also supports the [Koncorde Filters DSL](/core/2/api/koncorde-filters-syntax) to match documents by passing the `lang` argument with the value `koncorde`.  
+Koncorde filters will be translated into an Elasticsearch query.  
+
+::: warning
+Koncorde `bool` operator and `regexp` clause are not supported for search queries.
```
Comment (Leodau) [inline doc/1/controllers/document/update-by-query/index.md:20]:
```suggestion
Koncorde `bool` operator and `regexp` clauses are not supported for search queries.
```

File: doc/1/controllers/document/delete-by-query/index.md [hunk 5]
```diff
@@ -11,6 +11,15 @@ Deletes documents matching the provided search query.
 
 Kuzzle uses the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/query-dsl.html) syntax.
 
+<SinceBadge version="change-me"/>
+
+This method also supports the [Koncorde Filters DSL](/core/2/api/koncorde-filters-syntax) to match documents by passing the `lang` argument with the value `koncorde`.  
+Koncorde filters will be translated into an Elasticsearch query.  
+
+::: warning
+Koncorde `bool` operator and `regexp` clause are not supported for search queries.
```
Comment (Leodau) [inline doc/1/controllers/document/delete-by-query/index.md:20]:
```suggestion
Koncorde `bool` operator and `regexp` clauses are not supported for search queries.
```

File: .ci/doc/templates/default.tpl.java [hunk 6]
```diff
@@ -4,6 +4,7 @@
 import java.util.concurrent.ConcurrentHashMap;
 import io.kuzzle.sdk.coreClasses.responses.Response;
 import io.kuzzle.sdk.coreClasses.SearchResult;
+import io.kuzzle.sdk.coreClasses.lang.Lang;
```
Comment (Leodau) [inline .ci/doc/templates/default.tpl.java:7]: Isn't there any way we could beautify these paths :cry: `lang.Lang.lang.Lang`

**Review by Leodau — APPROVED** [review]
Mostly some lapsus plumæ

File: .ci/doc/templates/default.tpl.java (hunk 6 above)
Comment (Yoann-Abbes) [reply .ci/doc/templates/default.tpl.java:7]: `lang` is the name of the package (the folder) and `Lang` the class. I don't know :/ how to beautify it :/
Thats like `io.kuzzle.sdk.coreClasses.responses.Response;`

Comment (Leodau) [reply .ci/doc/templates/default.tpl.java:7]: Yep, I know. Just hoping there could be a way to get better paths, but well.

File: src/main/kotlin/io/kuzzle/sdk/controllers/DocumentController.kt [hunk 7]
```diff
@@ -94,20 +95,31 @@ class DocumentController(kuzzle: Kuzzle) : BaseController(kuzzle) {
...
       put("body", searchQuery)
       put("waitForRefresh", waitForRefresh)
+      put("lang", lang.lang)
     }
     return kuzzle
         .query(query)
         .thenApplyAsync { response -> (response.result as ConcurrentHashMap<String?, Any?>)["ids"] as ArrayList<String> }
   }
 
+  fun deleteByQuery(
```
Comment (jenow) [inline src/main/kotlin/io/kuzzle/sdk/controllers/DocumentController.kt:114]:
```suggestion
  @JvmOverloads
  fun deleteByQuery(
```

File: src/main/kotlin/io/kuzzle/sdk/controllers/DocumentController.kt [hunk 8]
```diff
@@ -361,13 +397,24 @@ class DocumentController(kuzzle: Kuzzle) : BaseController(kuzzle) {
...
       put("waitForRefresh", waitForRefresh)
+      put("lang", lang.lang)
     }
 
     return kuzzle
         .query(query)
         .thenApplyAsync { response -> response.result as ConcurrentHashMap<String, ArrayList<Any?>> }
   }
 
+  fun updateByQuery(
```
Comment (jenow) [inline src/main/kotlin/io/kuzzle/sdk/controllers/DocumentController.kt:408]:
```suggestion
  @JvmOverloads
  fun updateByQuery(
```

File: src/main/kotlin/io/kuzzle/sdk/coreClasses/lang/Lang.kt [hunk 9]
```diff
@@ -0,0 +1,6 @@
+package io.kuzzle.sdk.coreClasses.lang
+
+enum class Lang(val lang: String) {
+        KONCORDE("koncorde"),
+        ELASTICSEARCH("elasticsearch")
+}
```
Comment (jenow) [inline src/main/kotlin/io/kuzzle/sdk/coreClasses/lang/Lang.kt:6]:
```suggestion
}

```

File: src/main/kotlin/io/kuzzle/sdk/coreClasses/SearchResult.kt [hunk 10]
```diff
@@ -97,6 +100,6 @@ class SearchResult {
     val request: ConcurrentHashMap<String?, Any?> = nextRequest
     return kuzzle!!.query(nextRequest)
         .thenApplyAsync(
-            Function { response: Response -> SearchResult(kuzzle, request, scroll, from, size, response, fetched) })
+            Function { response: Response -> SearchResult(kuzzle, request, scroll, from, size, lang, response, fetched) })
   }
 }
```
Comment (jenow) [inline src/main/kotlin/io/kuzzle/sdk/coreClasses/SearchResult.kt:105]:
```suggestion
}

```

File: doc/1/controllers/document/update-by-query/snippets/update-by-query-koncorde-kotlin.kt [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+              },
+              {
+                _id=<document id>,
+                _source=<updated document> // if source set to true
+                status=200
+              }
+            ],
+    errors=[]
+}
+*/
```
Comment (jenow) [inline doc/1/controllers/document/update-by-query/snippets/update-by-query-koncorde-kotlin.kt:39]:
```suggestion
*/

```

File: doc/1/controllers/auth/search-api-keys/snippets/search-api-keys-koncorde-kotlin.kt [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+        "description"="LoRa 6 month API key",
+        "userId"="jared.doe",
+        "expiresAt"=31557600000,
+        "fingerprint": "4ee98cb8c614e99213e7695f822e42325d86c93cfaf39cb40e860939e784c8e6",
+        "ttl"=360000
+      }
+    }
+  ]
+}
+*/
```
Comment (jenow) [inline doc/1/controllers/auth/search-api-keys/snippets/search-api-keys-koncorde-kotlin.kt:72]:
```suggestion
*/

```

File: doc/1/controllers/document/delete-by-query/snippets/delete-by-query-koncorde-java.test.yml [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+    done
+    curl -XPOST kuzzle:7512/nyc-open-data/_refresh
+  after:
+template: print-result-arraylist
+expected:
+  - "document_1"
+  - "document_2"
+  - "document_3"
+  - "document_4"
+  - "document_5"
```
Comment (jenow) [inline doc/1/controllers/document/delete-by-query/snippets/delete-by-query-koncorde-java.test.yml:22]:
```suggestion
  - "document_5"

```

File: doc/1/controllers/document/delete-by-query/snippets/delete-by-query-koncorde-kotlin.test.yml [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+    done
+    curl -XPOST kuzzle:7512/nyc-open-data/_refresh
+  after:
+template: print-result-arraylist
+expected:
+  - "document_1"
+  - "document_2"
+  - "document_3"
+  - "document_4"
+  - "document_5"
```
Comment (jenow) [inline doc/1/controllers/document/delete-by-query/snippets/delete-by-query-koncorde-kotlin.test.yml:22]:
```suggestion
  - "document_5"

```

File: doc/1/controllers/document/update-by-query/snippets/update-by-query-koncorde-java.java [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+              },
+              {
+                _id=<document id>,
+                _source=<updated document> // if source set to true
+                status=200
+              }
+            ],
+    errors=[]
+}
+*/
```
Comment (jenow) [inline doc/1/controllers/document/update-by-query/snippets/update-by-query-koncorde-java.java:30]:
```suggestion
*/

```

File: doc/1/controllers/document/update-by-query/snippets/update-by-query-koncorde-java.test.yml [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+    done
+    for i in  1 2 3 4 5; do
+      curl -H "Content-type: application/json" -d '{"capacity": 7}' kuzzle:7512/nyc-open-data/yellow-taxi/_create
+    done
+    curl -XPOST kuzzle:7512/nyc-open-data/_refresh
+  after:
+template: print-result-array
+expected:
+  - "id=document_1(.*)_version=2(.*)status=200"
+  - "id=document_2(.*)_version=2(.*)status=200"
```
Comment (jenow) [inline doc/1/controllers/document/update-by-query/snippets/update-by-query-koncorde-java.test.yml:19]:
```suggestion
  - "id=document_2(.*)_version=2(.*)status=200"

```

**Review by jenow — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/kuzzle/sdk/controllers/DocumentController.kt (hunk 7 above)
Comment (Yoann-Abbes) [reply src/main/kotlin/io/kuzzle/sdk/controllers/DocumentController.kt:115]: This conflicts with the previous generated overloads

File: src/main/kotlin/io/kuzzle/sdk/controllers/DocumentController.kt (hunk 8 above)
Comment (Yoann-Abbes) [reply src/main/kotlin/io/kuzzle/sdk/controllers/DocumentController.kt:410]: same as above

**Review by scottinet — APPROVED** [review]

**Review by jenow — APPROVED** [review]

**Pull request merged**